lib/vmdb/global_methods.rb

```diff
@@ -36,31 +36,23 @@ def column_type(model, column)
     end
 
     # Had to add timezone methods here, they are being called from models
-    def get_timezone_abbr(typ="server")
-      # return timezone abbreviation
-      if typ == "server"
-        tz = MiqServer.my_server.get_config("vmdb").config.fetch_path(:server, :timezone)
-        tz = ActiveSupport::TimeZone::MAPPING[tz.blank? ? "UTC" : tz]
+    # return timezone abbreviation
+    def get_timezone_abbr(user = nil)
+      if user.nil?
+        tz = get_timezone_for_userid(nil)
         time = Time.now
       else
         tz = Time.zone
         time = Time.zone.now
       end
-      new_time = time.in_time_zone(tz)
-      abbr = new_time.strftime("%Z")
-      return abbr
+      time.in_time_zone(tz).strftime("%Z")
     end
 
-    def get_timezone_offset(typ="server",formatted=false)
-      # returns utc_offset of timezone
-      if typ == "server"
-        tz = MiqServer.my_server.get_config("vmdb").config.fetch_path(:server, :timezone)
-        tz = ActiveSupport::TimeZone::MAPPING[tz.blank? ? "UTC" : tz]
-      else
-        tz = get_timezone_for_userid(session[:userid])
-        @tz = tz unless tz.blank?
-        tz = ActiveSupport::TimeZone::MAPPING[@tz]
-      end
+    # returns utc_offset of timezone
+    def get_timezone_offset(user = nil, formatted = false)
+      tz = get_timezone_for_userid(user)
+      @tz = tz if user && tz.present?
```

Comment:
this logic was odd.
now, tz always has a value, so `present?` is not needed, but also note  `@tz` is only used in few places and may be dropped soon. If we keep it, this will need to follow standard memoization pattern